### ISSUE TITLE
PublicKey null, after connecting and ready

### ISSUE THREAD

**Issue by grumpyoldman-io** [solana-labs/wallet-adapter#31]
Im really liking this package, you take away a lot of the work but I've noticed a small bug / "growing pain" where the walletState is outputing the following:

```
{
  ...,
  connecting: false,
  connected: true,
  ready: true,
  publicKey: null
}
```

I would expect the publicKey to be set before the `connecting` flag is set to `false` after actually connecting. If you agree, I'd be happy to open a PR for this.

**Comment by jordansexton** [comment]
I agree... but I can't tell from your description where this is happening or how to reproduce this. Which package is this in?

**Comment by jordansexton** [comment]
I'm guessing it's this? https://github.com/solana-labs/wallet-adapter/blob/f14d5b30f7bdd2f47ee03c0744298f44b8803210/packages/react/src/WalletProvider.tsx#L70-L76

I think we may want to refactor this so we can do something like
```ts
const { connected, publicKey, autoApprove, ready } = adapter;
setState({ connected, publicKey, autoApprove, ready });
```
This will cause the state to be updated atomically to prevent inconsistencies.

**Comment by jordansexton** [comment]
@grumpyoldman-io please review: https://github.com/solana-labs/wallet-adapter/pull/33

**Comment by grumpyoldman-io** [comment]
Another solution would probably be to re-order the functions on line 73-75, but the proposed solution is way sturdier, at the cost of having a wee bit complexity in the state.

**Comment by jordansexton** [comment]
Fixed by #33
